releasing-archway.md

```diff
@@ -0,0 +1,111 @@
+# Archway Protocol Release Versioning
+
+Reference: [Semantic Versioning](http://semver.org)
+
+Legend:
+
+- **X.Y.Z** refers to the version (git tag) of Archway Protocol that is released.
+- **Network Operator** refers to an entity running a node and/or a validator and/or a relayer.
+- **archway-1** refers to the chain id of archway protocl main network.
+- **constantine-1** refers to the chain-id of the archway protocol canary public test network.
+- **titus-1** refers to the the chain-id of the archway protocol development public test network.
+
+Note: Please refer to https://github.com/archway-network/networks for a registry of all public networks running archway protocol
+and network specific information.
+
+## Release versioning
+
+### Minor version scheme and timeline
+- X.Y.Z-rc.W (Branch: release-X.Y)
+  - When main is feature-complete for X.Y, we will cut the release-X.Y
+    branch and cherrypick only PRs essential to X.Y.
+  - If we're not satisfied with X.Y.0-rc.0, we'll release other rc releases,
+    (X.Y.0-rc.W | W > 0) as necessary.
+- X.Y.0 (Branch: release-X.Y)
+  - Final release, cut from the release-X.Y branch cut two weeks prior.
+  - X.Y.1-rc.0 will be tagged at the same commit on the same branch.
+- X.Y.Z, Z > 0 (Branch: release-X.Y) ([Patch releases](#patch-releases))
+  - [Patch releases](#patch-releases) are released as we cherrypick commits from main into
+    the release-X.Y branch, as needed.
+  - X.Y.Z is cut straight from the release-X.Y branch, and X.Y.Z+1-beta.0 is
+    tagged on the followup commit that updates pkg/version/base.go with the beta
+    version.
+- X.Y.Z, Z > 0 (Branch: release-X.Y.Z) (Branched [patch releases](#patch-releases) only for hotfix situations)
+  - These are rarely used and are special in that the X.Y.Z tag is branched to isolate
+    the emergency/critical fix from all other changes that have landed on the
+    release branch since the previous tag
+  - Cut release-X.Y.Z branch to hold the isolated patch release
+  - Tag release-X.Y.Z branch + fixes with X.Y.(Z+1)
+  - Branched [patch releases](#patch-releases) are rarely needed but used for
+    emergency/critical fixes to the latest release
+
```

Comment:
^^^ X.Y.Z -> Breaking.Feature.Fix

This allows for clean monotonic version progression... 100% agree!

We can generally aim for a two week prior period, but this should be on a per release upgrade basis; we might need more time if we are dealing with breaking changes or complex state upgrade handlers;
We could set this as a minimum 2-week period...

Reply:
This proposal does not impose any time limits for release and the development team can decide on that from case to case basis.

Reply:
I agree, it does not make sense to put certain time slots on X. Y. Z

Reply:
L25?

Reply:
@zanicar good catch, it should be removed